web/src/components/dashboard/SidebarCardsList.test.tsx

```diff
@@ -62,4 +63,55 @@ describe("<SidebarCardsList />", () => {
     fireEvent.click(screen.getByTestId("hidden-opportunity-header"));
     expect(mockAnalytics.event.for_you_card_unhide_button.click.unhide_cards).toHaveBeenCalledTimes(1);
   });
+
+  describe("empty state messaging", () => {
+    it("displays when the For You section is empty", () => {
+      render(
+        <SidebarCardsList
+          topCards={[]}
+          bottomCards={[]}
+          fundings={[]}
+          hiddenFundings={[]}
+          certifications={[]}
+          hiddenCertifications={[]}
+          displayFundings={false}
+          displayCertifications={false}
+        />
+      );
+
+      expect(screen.getByTestId("empty-for-you-message")).toBeInTheDocument();
+      expect(screen.getByAltText("For You Section")).toBeInTheDocument();
+    });
+
+    it("does not display when the For You section has cards", () => {
+      const topCards = [
+        generateSidebarCardContent({
+          id: "some-fake-top-card",
+          section: "above-opportunities",
+        }),
+      ];
+
+      const bottomCards = [
+        generateSidebarCardContent({
+          id: "some-fake-bottom-card",
+          section: "below-opportunities",
+        }),
+      ];
+      render(
+        <SidebarCardsList
+          topCards={topCards}
+          bottomCards={bottomCards}
```

Comment:
[pebble] It's generally good practice to test edge cases. In these two tests we have one test that says "when there are 0 cards, show the Empty State Message" and one test that says "when there are 2 cards, do not show the Empty State Message"

It leaves the 1 case untested. Generally it's better to test the case right on the edge of the logic instead of something in the middle (like 2). I might recommend writing this test as only 1 card instead of 2 would be more useful.

Think of it like the AQI TQ example. If you're testing that an aqi>100 is "unhealthy" and less than 100 is "good" then you could write tests like:
```
expect(getAQI(50)).toEqual("good")
expect(getAQI(125)).toEqual("unhealthy")
```

This is correct testing, but the logic could vary and the test would still pass. Any of the following lines of code would get the test passing:
```
return aqi > 100 ? "unhealthy" : "good"
return aqi >= 100 ? "unhealthy" : "good"
return aqi > 80 ? "unhealthy" : "good"
return aqi > 120 ? "unhealthy" : "good"
```

So the tests, while accurate, don't actually test that the boundary line is 100 specifically. A better test would test the specific edge cases:
```
expect(getAQI(99)).toEqual("good")
expect(getAQI(100)).toEqual("good")
expect(getAQI(101)).toEqual("unhealthy")
```

the **only** line of code that would allow this to pass is:
```
return aqi >= 100 ? "unhealthy" : "good"
```

This is why we test boundary conditions. So I think this test could be improved by testing what happens when we have _one_ card in the For You column (instead of 2)